### PR TITLE
Feature/add to queue

### PR DESCRIPTION
**Pull request by xHawkyx**
pls be gentle...

### CODE REVIEW

File: src/Components/SubListComponent.ts [hunk 1]
```diff
@@ -52,6 +59,15 @@ export default class SubListComponent extends Component {
         });
     }
 
+    startQueue(): void {
+        if (queueService.getStartVideoId() == null){
+            alert("Queue is empty, you need to add videos first, duh!");
```
Comment (Crow08) [inline src/Components/SubListComponent.ts:64]: do you even add bro!

File: src/Components/VideoComponent.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import moment from "moment";
+import { isPropertyName } from "typescript";
```
Comment (Crow08) [inline src/Components/VideoComponent.ts:2]: unused import

File: src/Components/VideoComponent.ts [hunk 3]
```diff
@@ -35,14 +38,19 @@ export default class VideoComponent extends Component {
         });
         this.clicksItem = $("<p/>", {"class": "ytbsp-views", "html": video.clicks});
         this.uploadItem = $("<p/>", {"class": "ytbsp-uploaded", "html": video.uploaded});
-        this.seenMarkerItem = $("<p/>", {
+        this.addToQueueItem = $("<p/>", {
```
Comment (Crow08) [inline src/Components/VideoComponent.ts:41]: can also be a span instead of p

File: src/Components/VideoComponent.ts [hunk 4]
```diff
@@ -35,14 +38,19 @@ export default class VideoComponent extends Component {
...
+        });
+        this.seenMarkerItem = $("<span/>", {
             "class": `ytbsp-seenMarker${video.seen ? " seen" : ""}`,
             "html": (video.seen ? "already seen" : "mark as seen")
         });
 
         this.clipItem.mouseover(() => this.startEnlargeTimeout());
         this.clipItem.mouseleave(() => this.abortEnlargeTimeout());
         this.closeItem.mouseover(() => this.abortEnlargeTimeout());
+        this.addToQueueItem.mouseover(() => this.abortEnlargeTimeout());
```
Comment (Crow08) [inline src/Components/VideoComponent.ts:53]: this change is not needed anymore.

File: src/Components/VideoComponent.ts [hunk 5]
```diff
@@ -67,8 +77,17 @@ export default class VideoComponent extends Component {
...
+        //this adds clicked video to a magical invisible youtube playlist
+        this.addToQueueItem.click(() => {
+            pageService.addToQueue(video.id);
+            queueService.setStartVideoId(video.id);
+            this.addToQueueItem.css("color","green");
+            this.addToQueueItem.html("ADDED &#10003;");
+        });
 
         this.clipItem.add(this.titleItem).add(this.closeItem).click((event) => this.handleOpenVideo(event));
+        this.clipItem.add(this.titleItem).add(this.addToQueueItem).click((event) => this.handleOpenVideo(event));
```
Comment (Crow08) [inline src/Components/VideoComponent.ts:90]: this change is not needed anymore.

**Review by Crow08 — CHANGES_REQUESTED** [review]

File: src/Components/VideoComponent.ts (hunk 2 above)
Comment (xHawkyx) [reply src/Components/VideoComponent.ts:2]: removed useless import

File: src/Components/SubListComponent.ts (hunk 1 above)
Comment (xHawkyx) [reply src/Components/SubListComponent.ts:64]: added rickroll, changed alert a bit

File: src/Components/VideoComponent.ts (hunk 3 above)
Comment (xHawkyx) [reply src/Components/VideoComponent.ts:41]: changed to <span>

File: src/Components/VideoComponent.ts [hunk 6]
```diff
@@ -153,7 +172,7 @@ export default class VideoComponent extends Component {
 
     private handleOpenVideo(event: ClickEvent): void {
         event.preventDefault();
-        if ((event.target as Element).classList.contains(this.closeItem.attr("class"))) {
+        if ((event.target as Element).classList.contains(this.closeItem.attr("class")) || (event.target as Element).classList.contains(this.addToQueueItem.attr("class"))) {
```
[the remark this answers is not in the document]
Comment (xHawkyx) [reply src/Components/VideoComponent.ts:174]: removed useless code

File: src/Components/VideoComponent.ts (hunk 4 above)
Comment (xHawkyx) [reply src/Components/VideoComponent.ts:52]: removed shamefull code

File: src/Components/VideoComponent.ts (hunk 5 above)
Comment (xHawkyx) [reply src/Components/VideoComponent.ts:89]: removed preposterous line of code

**Review by Crow08 — APPROVED** [review]

**Pull request merged**